src/Codeception/Module/REST.php

```diff
@@ -333,15 +334,35 @@ public function amNTLMAuthenticated($username, $password)
     }
 
     /**
-     * Adds AWS Authentication via key, secret, service and region
-     * @param $key
-     * @param $secret
-     * @param $service
-     * @param $region
+     * Allows to send REST request using AWS Authorization
+     * Requires client to by Guzzle
+     * Example
+     * Config -
+     *
+     * modules:
+     *      enabled:
+     *          - REST:
+     *              aws:
+     *                  key: accessKey
+     *                  secret: accessSecret
+     *                  service: awsService
+     *                  region: awsRegion
+     *
+     * ```php
+     * <?php
+     * $I->amAWSAuthenticated();
+     * ?>
+     * ```
+     * @param array $additionalAWSConfig
+     * @throws ModuleException
      */
-    public function amAWSAuthenticated($key, $secret, $service, $region)
+    public function amAWSAuthenticated($additionalAWSConfig = [])
     {
-        $this->client->setAwsAuth($key, $secret, $service, $region);
+        if (!defined('\GuzzleHttp\Client::VERSION')) {
```

Comment:
This condition is wrong. Guzzle is a mandatory dependency - so the only case when this exception could be thrown is if Guzzle 5 or 4 is installed.
`method_exists($this->client, 'setAwsAuth')` or `$this->client instanceof \Codeception\Lib\Connector\Guzzle6` are better ways.

Also I don't like `Out of scope` in exception message.

Reply:
Done